### ISSUE TITLE
GET Requests: /federations/{{federationsId}}/deliveryservices success when Federation doesn't exist 

### ISSUE THREAD

**Issue by dpham692** [apache/trafficcontrol#4245]
## I'm submitting a ...

- [x] bug report
- [ ] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:

Do not create a Federations.
GET Request: 

`https://{{TO_BASE_URL}}/api/{{api_version}}/federations/1231/deliveryservices`

It will return 200 OK
```
{
    "response": []
}
```
## Expected / new behavior:

Should return 404 Not Found

**Comment by mitchell852** [comment]
@dpham692 - does a federation with id=1231 exist?

**Comment by mitchell852** [comment]
yeah, if a federation with id=1231 does not exist then a 404 would be the correct response.
